frontend/gamelib.js

```diff
@@ -244,26 +247,29 @@ class Region {
 		return Calc.moddedNum(base, mods);
 	}
 
-	calcHarvest() {
+	calcCrops() {
+		return Calc.moddedNum({"v": this.crops, "unit": " crops", "why": "Already Planted"}, []);
+	}
+
+	calcHarvestCapacity() {
 		let base = new Calc("*", [
 			{"v": this.population, "unit": " citizens", "why": "Regional Population"},
-			{"v": this.harvest, "unit": " measures / citizen", "why": "Regional Havest Rate"}]);
+			{"v": 25, "unit": " reaps / citizen", "why": "Global Havest Rate"}]);
```

Comment:
Just a comment, this line leads me to believe that all of the Constants we've started amassing should eventually become the template for a game config file that can be used across backend and frontend as the single source of truth, easy to tweak modifiers that way by limiting the blast radius of a change.

Reply:
Yes exactly! I am working on a change along these lines in another branch. The basic idea is to define a directory structure like:

ruleset_id (1, 2, 3, etc)
  \_ rules.json
  \_ setup.json
  \_ maps
      \_ 2_player.json
      \_ 3_player.json
      \_ ...
      \_ 26_player.json

remove WorldConstantData.java, serve this data from the backend, and pull it into the frontend. 

Of course, that being said, in practice a lot of the rule changes are actually additions/removals of mechanics or formulae, not tweaks of numeric constants. So that will continue to be challenging. I don't yet see a way to define a generic common grammer to completely describe the game consistently to both the frontend and backend. But even so, longer-term I like the idea of running multiple games concurrently on slightly different rulesets (if for no other reason that I want multiple world setups, and scoring is only comparable within a particular ruleset).